### PR TITLE
Docs: add documentation for `GroupPath`

### LINKED ISSUE #3913: Document New Group Utilities
Once the new group based capabilities are finalized, principally #3613, #3882, #3903,
we should update the documentation, e.g. `docs/source/working_with_aiida/groups.rst`,
with a guide for users.

### PR DESCRIPTION
**Pull request by chrisjsewell**
closes #3913

### CODE REVIEW

File: docs/source/working_with_aiida/groups.rst [hunk 1]
```diff
@@ -177,3 +177,108 @@ This is what AiiDA uses to load the correct class when reloading the group from
...
+
+    In [12]: path["base1/sub_group1"].delete_group()
+    In [13]: path["base1/sub_group1"].is_virtual
+    Out[13]: True
+    In [14]: path["base1/sub_group1"].get_or_create_group()
+    Out[14]: (<Group: "base1/sub_group1" [type core], of user user@email.com>, True)
+    In [15]: path["base1/sub_group1"].is_virtual
+    Out[15]: False
+
+To traverse paths, the `children` attribute can be used, or for recursive traversal::
```
Comment (ltalirz) [inline docs/source/working_with_aiida/groups.rst:256]:
```suggestion
To traverse paths, use the `children` attribute - for recursive traversal, use `walk`::
```

File: docs/source/working_with_aiida/groups.rst [hunk 2]
```diff
@@ -177,3 +177,108 @@ This is what AiiDA uses to load the correct class when reloading the group from
...
+or the group can be retrieved with the `get_group()` method::
+
+    In [8]: path.is_virtual
+    Out[8]: True
+    In [9]: path.get_group() is None
+    Out[9]: True
+    In [10]: path["base1/sub_group1"].get_group() is None
+    Out[10]: True
+    In [11]: path["base1/sub_group1"].get_group()
+    Out[11]: <Group: "base1/sub_group1" [type core], of user user@email.com>
```
Comment (sphuber) [inline docs/source/working_with_aiida/groups.rst:244]: As @mbercx highlighted, these two results seen contradicting no?

Comment (chrisjsewell) [reply docs/source/working_with_aiida/groups.rst]: yep copy-paste error 😬 fixed in 9b2f222

**Comment by chrisjsewell** [comment]
> Please maintain one sentence per line for text.

Do you mean converting this:

```
You can also traverse directly through the nodes of a path,
optionally filtering by node class and any other filters allowed by the `QueryBuilder`::
```

to this:

```
You can also traverse directly through the nodes of a path, optionally filtering by node class and any other filters allowed by the `QueryBuilder`::
```

because you end up with quite long lines, which is why I'd added a few breaks on commas.

**Comment by chrisjsewell** [comment]
> Could you please add a sentence or two on the function of cls in the GroupPath and the current limitations?

Added in 506a6c6

**Comment by CasperWA** [comment]
Should this be added before revamping the docs?

**Comment by chrisjsewell** [comment]
I think so. Its ready for merging IMO

**Comment by sphuber** [comment]
> because you end up with quite long lines, which is why I'd added a few breaks on commas.

Yes, that is what I mean, because that is the [style rule that was decided](https://github.com/aiidateam/aiida-core/wiki/Writing-documentation#style-guide). We should not add arbitrary line breaks but just let the editor use soft line-wrapping.

Regarding the merging. Since anyway this will have to be moved once we have the new skeleton, there is no rush right? Since it is purely adding new text, there won't be any merge conflicts. So I actually would hold off with merging this until the skeleton is inplace, and then we can add it in the correct place straight away. Doing the same with my open docs PR #3966

**Comment by chrisjsewell** [comment]
> Please maintain one sentence per line for text.

Fixed in 3cc9a03

**Comment by CasperWA** [comment]
> Regarding the merging. Since anyway this will have to be moved once we have the new skeleton, there is no rush right? Since it is purely adding new text, there won't be any merge conflicts. So I actually would hold off with merging this until the skeleton is inplace, and then we can add it in the correct place straight away. Doing the same with my open docs PR #3966

Fine. Then I'll add it to the project instead.

**Comment by csadorf** [comment]
@chrisjsewell Could you adapt this PR so that we can merge it into docs-revamp?

**Comment by chrisjsewell** [comment]
> Could you adapt this PR so that we can merge it into docs-revamp?

On it now 👍

**Comment by chrisjsewell** [comment]
rebased to #4126

**Pull request closed without merging**